packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -148,12 +156,26 @@ class _SegmentState<T> extends State<_Segment<T>> with TickerProviderStateMixin<
     assert(oldWidget.key == widget.key);
 
     if (oldWidget.shouldScaleContent != widget.shouldScaleContent) {
-      highlightPressScaleAnimation = highlightPressScaleController.drive(
-        Tween<double>(
-          begin: highlightPressScaleAnimation.value,
-          end: widget.shouldScaleContent ? _kMinThumbScale : 1.0,
-        ),
-      );
+      final Animatable<double> scaleAnimation =
+          widget.isMomentary && widget.shouldScaleContent
```

Comment:
Seems `widget.shouldScaleContent` will check `isMomentary`, so we can remove `widget.isMomentary` check here?

Reply:
No, I think because shouldScaleContent can be true even when isMomentary is false.

Reply:
Oh that makes sense. Thank you for your explanation!